src/htm/regions/SPRegion.cpp

```diff
@@ -611,10 +616,27 @@ UInt32 SPRegion::getParameterUInt32(const std::string &name, Int64 index) {
     break;
   case 'n':
     if (name == "numActiveColumnsPerInhArea") {
-      if (sp_)
-        return sp_->getNumActiveColumnsPerInhArea();
-      else
-        return args_.numActiveColumnsPerInhArea;
+      NTA_WARN << "SPRegion: `numActiveColumnsPerInhArea` is deprecated, use `localAreaDensity` instead.";
+      Real density = -1.0f;
+      UInt inhRadius = -1u;
+      vector<UInt> dims;
+      if (sp_ != nullptr) {
+        density = sp_->getLocalAreaDensity();
+        inhRadius = sp_->getInhibitionRadius();
+	dims = sp_->getColumnDimensions();
+      } else {
+        density = args_.localAreaDensity;
+	NTA_THROW << "Cannot compute from these params!";
```

Comment:
this still fails if `sp_` is not initialized. As those params I need to access (inhibitionRadius_, columnDimensions_) are not part of args_. 
Otherwise this works, so I guess it's OK to keep like that considering the whole param is strongly deprecated.